lucene/core/src/java/org/apache/lucene/search/BlockMaxConjunctionBulkScorer.java

```diff
@@ -68,18 +88,10 @@ public int score(LeafCollector collector, Bits acceptDocs, int min, int max) thr
       // Use impacts of the least costly scorer to compute windows
       // NOTE: windowMax is inclusive
       int windowMax = Math.min(scorers[0].advanceShallow(windowMin), max - 1);
-      for (int i = 1; i < scorers.length; ++i) {
-        scorers[i].advanceShallow(windowMin);
-      }
 
-      double maxWindowScore = 0;
-      for (int i = 0; i < scorers.length; ++i) {
-        double maxClauseScore = scorers[i].getMaxScore(windowMax);
-        sumOfOtherClauses[i] = maxClauseScore;
-        maxWindowScore += maxClauseScore;
-      }
-      for (int i = sumOfOtherClauses.length - 2; i >= 0; --i) {
-        sumOfOtherClauses[i] += sumOfOtherClauses[i + 1];
+      double maxWindowScore = Double.POSITIVE_INFINITY;
```

Comment:
We already cast before returning in `computeMaxScore`, so no need to make it a `double` here only to cast again on when calling `scoreWindow`.

```suggestion
      float maxWindowScore = Float.POSITIVE_INFINITY;
```